src/integration-test/java/com/mongodb/jdbc/integration/testharness/TestGenerator.java

```diff
@@ -0,0 +1,145 @@
+            expectedIsCaseSensitive.add(resultSetMetadata.isCaseSensitive(i));
+            expectedIsCurrency.add(resultSetMetadata.isCurrency(i));
+            expectedIsDefinitelyWritable.add(resultSetMetadata.isDefinitelyWritable(i));
+            expectedIsNullable.add(
+                    TestTypeInfo.nullableIntToString(resultSetMetadata.isNullable(i)));
+            expectedIsReadOnly.add(resultSetMetadata.isReadOnly(i));
+            expectedIsSearchable.add(resultSetMetadata.isSearchable(i));
+            expectedIsSigned.add(resultSetMetadata.isSigned(i));
+            expectedIsWritable.add(resultSetMetadata.isWritable(i));
+        }
+
+        testCase.put("description", testEntry.description);
+        if (testEntry.meta_function != null) {
+            testCase.put("meta_function", testEntry.meta_function);
+        } else {
+            testCase.put("sql", testEntry.sql);
+        }
+        testCase.put("expected_result", result);
+        testCase.put("row_count", testEntry.row_count);
+        testCase.put("row_count_gte", testEntry.row_count_gte);
+        testCase.put("ordered", testEntry.ordered);
+
+        testCase.put("expected_sql_type", expectedSqlType);
+        testCase.put("expected_catalog_name", expectedCatalogName);
+        testCase.put("expected_column_class_name", expectedColumnClassName);
+        testCase.put("expected_column_label", expectedColumnLabel);
+        testCase.put("expected_column_display_size", expectedColumnDisplaySize);
+        testCase.put("expected_precision", expectedPrecision);
+        testCase.put("expected_scale", expectedScale);
+        testCase.put("expected_schema_name", expectedSchemaName);
+        testCase.put("expected_is_auto_increment", expectedIsAutoIncrement);
+        testCase.put("expected_is_case_sensitive", expectedIsCaseSensitive);
+        testCase.put("expected_is_currency", expectedIsCurrency);
+        testCase.put("expected_is_definitely_writable", expectedIsDefinitelyWritable);
+        testCase.put("expected_is_nullable", expectedIsNullable);
+        testCase.put("expected_is_read_only", expectedIsReadOnly);
+        testCase.put("expected_is_searchable", expectedIsSearchable);
+        testCase.put("expected_is_signed", expectedIsSigned);
+        testCase.put("expected_is_writable", expectedIsWritable);
+
+        DumperOptions options = new DumperOptions();
+        options.setIndent(2);
+        options.setIndicatorIndent(2);
+        options.setIndentWithIndicator(true);
+        Yaml yaml = new Yaml(options);
+        FileWriter writer = new FileWriter(GENERATED_TEST_DIR + "/" + fileName);
+        yaml.dump(tests, writer);
+    }
+
+    public static void main(String[] args)
+            throws SQLException, IOException, InvocationTargetException, IllegalAccessException {
+        MongoSQLIntegrationTest integrationTest = new MongoSQLIntegrationTest();
+        Connection conn = integrationTest.getBasicConnection();
```

Comment:
Looks like you missed closing the connection here. Can you add a try with-resource here?